docker-compose.yml

```diff
@@ -296,7 +296,17 @@ services:
   spark-service:
     image: "quay.io/cdis/gen3-spark:2021.03"
     container_name: spark-service
-    command: bash -c "python run_config.py && hdfs namenode -format && hdfs --daemon start namenode && hdfs --daemon start datanode && yarn --daemon start resourcemanager && yarn --daemon start nodemanager && hdfs dfsadmin -safemode leave &&  hdfs dfs -mkdir /result && while true; do sleep 5; done"
+    command: >
+      bash -c "
+      python run_config.py
+      && ([ -f /hadoop/hdfs/data/dfs/namenode/current/VERSION ] || hdfs namenode -format)
```

Comment:
This works in that the errors...

```
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
spark-service        | Re-format filesystem in Storage Directory root= /hadoop/hdfs/data/dfs/namenode; location= null ? (Y or N) Invalid input:
```

...go away, but I don't know how best to test this in reality.

Reply:
I'm actually surprised we don't see users talking about that issue in the wild? Maybe there's something about most user configs that this isn't a problem? Today I can easily recreate this if I start fresh, bring compose down, then bring it back up. Not sure if that's just par for the course for others or some condition that I should be looking towards that could be valid with the original command.